### ISSUE TITLE
Upgrade to argo-cd 2.3.x

### ISSUE THREAD

**Issue by davidkarlsen** [redhat-developer/gitops-operator#285]
**Is your feature request related to a problem? Please describe.**
Upgrade to latest 2.3.x which contains many enhancements: https://github.com/argoproj/argo-cd/releases/tag/v2.3.0

**Describe the solution you'd like**
Use v2.3.0

**Describe alternatives you've considered**
N/A

**Additional context**
https://blog.argoproj.io/argo-cd-v2-3-release-candidate-a5b8cf11b0d3

**Comment by redox72** [comment]
Huge thumbs up for that.
Out of the box ApplicationSet is something I am waiting for

**Comment by greglanthier** [comment]
[argo-cd pr#3959](https://github.com/argoproj/argo-cd/pull/3959) - which merged in time for [argo-cd v2.3.1](https://github.com/argoproj/argo-cd/releases/tag/v2.3.1) - has a fix for resource deletion that honours sync-wave ordering that would be very useful.

There are scenarios where you might want to do a cascading delete of an ApplicationSet or an App of apps collection that contains a resource with a finalizer along with the operator that is responsible for processing that finalizer.

### For Example

I have an ApplicationSet that installs one ArgoCD application that does cluster-setup - including installing the [ibmcloud-operator](https://github.com/IBM/cloud-operators) - and another ArgoCD application that creates a resource managed by the **ibmcloud-operator**.

![image](https://user-images.githubusercontent.com/324373/159139136-53658a29-1765-4dec-b7b0-3fca9d460fb4.png)

The **mongodb** resource has a spec that looks like this:

```yaml
apiVersion: ibmcloud.ibm.com/v1
kind: Service
metadata:
  ...
  finalizers:
  - service.ibmcloud.ibm.com
  name: mongodb
  namespace: databases
spec:
  ...
```

Prior to [PR#3959](https://github.com/argoproj/argo-cd/pull/3959) when ArgoCD performs a cascading delete the order resources are removed is undefined.

This means with ArgoCD versions prior to 2.3.1 in the example above the **ibmcloud-operator** might be deleted first which would prevent the finalizer in the **mongodb** resource from being processed - and so Kubernetes would not be able to delete the resource on its own without intervention.

Using sync-waves to control the order of installing _and_ deleting resources would help workaround these types of sequencing problems.

**Comment by jannfis** [comment]
Hi folks, sorry for the late reply. 

Argo CD v2.3 will be shipped with GitOps Operator v1.5.0. We plan to ship v1.5.0 on March 30th, but as always, this date may change.

**Comment by iam-veeramalla** [comment]
GitOps operator v1.5.0 is out.
